### PR TITLE
feat: Support temporary tables

### PR DESCRIPTION
**Pull request by vrongmeal**
A couple of things to note here, I haven't used the `CatalogProvider` or `SchemaProvider`, instead just resolved the table myself. This is technically what is happening internally but there's more projection stuff that I am missing out on which makes the following query fail:

```
glaredb=> insert into abc (a, c) values (2, 4.678);
ERROR:  Error during planning: Inserting query must have the same schema with the table.
```

If we insert into all columns, this works:

```
glaredb=> insert into abc (a, b, c) values (2, 'abc', 4.678);
INSERT
```

Also need to refine `glare_catalog.tables` result (add oid there).

### CODE REVIEW

**Comment by scsmithr** [comment]
> I haven't used the CatalogProvider or SchemaProvider, instead just resolved the table myself. 

I think this is for the best. We're not using these elsewhere, so I think it's fine that we're not using them here.

File: crates/sqlexec/src/planner/dispatch.rs [hunk 1]
```diff
@@ -649,6 +659,19 @@ impl<'a> SystemTableDispatcher<'a> {
             datasource.append_value(table.options.as_str());
         }
 
+        // Append temporary tables.
+        for table in self.ctx.list_temp_tables() {
+            // TODO: Assign OID to temporary tables
```
Comment (scsmithr) [inline crates/sqlexec/src/planner/dispatch.rs:664]: I don't know if we'll want or need to assign oids to these tables. Going to check out what some other systems do.

**Review by scsmithr — COMMENTED** [review]
So far looks good to me.

**Comment by vrongmeal** [comment]
As far as I can see, DF doesn't support `insert into table (a, b) values (1, 2)` where the table has a different schema (yes, even the ordering of fields matters here).

I'll see if I can get a wrapper over `ExecutionPlan` to make it work. We have the schema, so it shouldn't be an issue. DF ignores the schema right now

File: crates/sqlexec/src/planner/dispatch.rs (hunk 1 above)
Comment (scsmithr) [reply crates/sqlexec/src/planner/dispatch.rs:664]: Yep, seems like duckdb does too. I think we go with 0 for now. We'll probably want to extend the `SessionCatalog` eventually to support the temp stuff.

**Comment by vrongmeal** [comment]
Removed most of "halts" (which were due to lack of `create table` and `insert into`). Couldn't get `insert into ABC (d, e, f)` working easily. Will look into it. The tests don't rely on it though :)

**Pull request merged**